### ISSUE TITLE
Revise segfaults only in Juno

### ISSUE THREAD

**Issue by ianfiske** [timholy/Revise.jl#319]
After updating Juno and several packages, I started getting the segfault
```
signal (11): Segmentation fault: 11
in expression starting at /Users/ifiske/.julia/packages/Revise/agmgx/test/runtests.jl:69
_ZN12_GLOBAL__N_114CodeGenPrepare13runOnFunctionERN4llvm8FunctionE at /Applications/Julia-1.1.app/Contents/Resources/julia/lib/julia/libLLVM.dylib (unknown line)
_ZN4llvm13FPPassManager13runOnFunctionERNS_8FunctionE at /Applications/Julia-1.1.app/Contents/Resources/julia/lib/julia/libLLVM.dylib (unknown line)
_ZN4llvm13FPPassManager11runOnModuleERNS_6ModuleE at /Applications/Julia-1.1.app/Contents/Resources/julia/lib/julia/libLLVM.dylib (unknown line)
_ZN4llvm6legacy15PassManagerImpl3runERNS_6ModuleE at /Applications/Julia-1.1.app/Contents/Resources/julia/lib/julia/libLLVM.dylib (unknown line)
```
immediately after changing any code and evaluating something in the REPL.

I have Revise configured to start on Julia-startup as in the Revise docs.  As an MWE, this occurs as soon as the tests begin in a fresh Julia session:
```
Starting Julia...
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.1.1 (2019-05-16)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(v1.1) pkg> test Revise
   Testing Revise
 Resolving package versions...
    Status `/var/folders/zz/mlzd8jsj2nsfz5bq2w3m21r000043q/T/tmpLaxk25/Manifest.toml`
  [da1fd8a2] CodeTracking v0.5.7
  [97e2ac4a] EponymTuples v0.2.2
  [7876af07] Example v0.5.1
  [aa1ae85d] JuliaInterpreter v0.6.0
  [6f1432cf] LoweredCodeUtils v0.3.5
  [bac558e1] OrderedCollections v1.1.0
  [295af30f] Revise v2.1.6
  [2a0f44e3] Base64  [`@stdlib/Base64`]
  [ade2ca70] Dates  [`@stdlib/Dates`]
  [8ba89e20] Distributed  [`@stdlib/Distributed`]
  [7b1f6079] FileWatching  [`@stdlib/FileWatching`]
  [b77e0a4c] InteractiveUtils  [`@stdlib/InteractiveUtils`]
  [76f85450] LibGit2  [`@stdlib/LibGit2`]
  [56ddb016] Logging  [`@stdlib/Logging`]
  [d6f4376e] Markdown  [`@stdlib/Markdown`]
  [44cfe95a] Pkg  [`@stdlib/Pkg`]
  [de0858da] Printf  [`@stdlib/Printf`]
  [3fa0cd96] REPL  [`@stdlib/REPL`]
  [9a3f8284] Random  [`@stdlib/Random`]
  [ea8e919c] SHA  [`@stdlib/SHA`]
  [9e88b42a] Serialization  [`@stdlib/Serialization`]
  [6462fe0b] Sockets  [`@stdlib/Sockets`]
  [8dfed614] Test  [`@stdlib/Test`]
  [cf7118a7] UUIDs  [`@stdlib/UUIDs`]
  [4ec0a83e] Unicode  [`@stdlib/Unicode`]
Skipping Base.active_repl
Skipping Base.active_repl_backend

signal (11): Segmentation fault: 11
in expression starting at /Users/ifiske/.julia/packages/Revise/agmgx/test/runtests.jl:69
_ZN12_GLOBAL__N_114CodeGenPrepare13runOnFunctionERN4llvm8FunctionE at /Applications/Julia-1.1.app/Contents/Resources/julia/lib/julia/libLLVM.dylib (unknown line)
_ZN4llvm13FPPassManager13runOnFunctionERNS_8FunctionE at /Applications/Julia-1.1.app/Contents/Resources/julia/lib/julia/libLLVM.dylib (unknown line)
_ZN4llvm13FPPassManager11runOnModuleERNS_6ModuleE at /Applications/Julia-1.1.app/Contents/Resources/julia/lib/julia/libLLVM.dylib (unknown line)
_ZN4llvm6legacy15PassManagerImpl3runERNS_6ModuleE at /Applications/Julia-1.1.app/Contents/Resources/julia/lib/julia/libLLVM.dylib (unknown line)
```

The tests run successfully and I do not get these crashes when running in the terminal outside of Juno.

Things I have tried without success include

- deleting all of ~/.julia/packages and ~/.julia/compiled followed by `] resolve/instantiate/build/compile`
- restarting Atom
- ensuring all julia processes are killed
- rebooting
- upgrading from Julia 1.1 to 1.1.1

System info:
```
julia> versioninfo()
Julia Version 1.1.1
Commit 55e36cc (2019-05-16 04:10 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin15.6.0)
  CPU: Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
Environment:
  JULIA_NUM_THREADS = 4
```

**Comment by ianfiske** [comment]
This seems to be the same as https://discourse.julialang.org/t/segmentation-fault-when-updating-packages/23766

**Comment by ianfiske** [comment]
This issue was resolved by uninstalling Juno and doing a fresh install.  Revise tests now pass in Juno and it works for me in general.  My old Juno install was an Atom install that was included with an old JuliaPro, like 6.x.  I thought it was kept up-to-date through the in-Atom updates and I had been using the same Atom through all Julia updates through the current 1.1.1, but apparently something must have got out-of-sync or outdated.  Anyhow, for others' reference, I also did `rm -rf ~/.atom` to completely start fresh before the new Atom install.
